lib/src/outputs/advanced_file_output_stub.dart

```diff
@@ -0,0 +1,29 @@
+import 'dart:convert';
+import 'dart:io';
+
+import '../log_level.dart';
+import '../log_output.dart';
+import '../output_event.dart';
+
+/// Writes the log output to a file.
+class AdvancedFileOutput extends LogOutput {
+  AdvancedFileOutput({
+    Directory? directory,
+    File? file,
+    bool overrideExisting = false,
+    Encoding encoding = utf8,
+    List<Level>? writeImmediately,
+    Duration maxDelay = const Duration(seconds: 2),
+    int maxBufferSize = 2000,
+    int maxLogFileSizeMB = 1,
+  }) {
+    throw UnsupportedError("Not supported on this platform.");
+  }
+
+  File? get targetFile => null;
```

Comment:
Some of your public fields are missing here (means they can't really be accessed by the developer).
Please either make them private or provide proper getters for them. (Note: Getters can throw `UnsupportedErrors` too!)

Reply:
Okay. I think I'll make most of them private because changing them after logger creation may result in unexpected behaviour + it's generally hard to image why someone will need getters for these things...